fitbenchmarking/fitbenchmarking
# Algorithm type option

#### Description of Work

Fixes #525 

Adds an algorithm type option which the user can use to select the type of algorithms used.

Within this PR I notice that the tables created an error if all minimizers raised and exception. I have created an optional argument for the table template which would report the specific problem sets which these failed on. Together with this, if the minimizer is not selected due to the algorithm type option, tables will be produced without this as a column and it will be reported below in a separate section


#### Testing Instructions

1. Check that the `algorithm_type` option works as expected
2. Check that the controllers dictionary which selects the types are correct

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?
